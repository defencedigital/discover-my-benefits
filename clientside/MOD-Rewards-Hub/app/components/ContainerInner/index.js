import React from 'react';
import PropTypes from 'prop-types';

/**
 * @summary Container Inner
 * @param {{ children: JSX.Element}} [children]
 * @property {string} className classname to be passed in
 */

class ContainerInner extends React.PureComponent {
  render() {
    const { children, className } = this.props;

    return <div className={`container-inner ${className || ''}`}>{children}</div>;
  }
}

ContainerInner.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  className: PropTypes.string,
};

export default ContainerInner;
