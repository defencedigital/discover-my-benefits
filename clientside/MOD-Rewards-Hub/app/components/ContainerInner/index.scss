.container-inner {
  padding: 0 20px;

  @include media-breakpoint-down(md) {
    padding: 0 10px;
  }

  @media all and (min-width: 1120px) {
    max-width: 1280px;
    margin: 0 auto;

    &.no-padding-lrg {
      padding: 0;
    }
    &.modal-padding-top {
      padding-top: 70px;
    }
  }

  &.profile-container {
    .form {
      padding: 24px 0;
    }
  }
}

.container-inner + .footer {
  margin-top: 50px;
}
