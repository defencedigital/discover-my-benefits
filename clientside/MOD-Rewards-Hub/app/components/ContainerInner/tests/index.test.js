import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ContainerInner from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  class: 'class',
  children: ['<div>sf</div>'],
});

describe('ContainerInner component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<ContainerInner {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
