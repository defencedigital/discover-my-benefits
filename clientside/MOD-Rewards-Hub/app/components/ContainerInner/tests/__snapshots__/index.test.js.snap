// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerInner component matches the snapshot 1`] = `ShallowWrapper {}`;
